### PR TITLE
Add unit tests for the `events.py` module

### PR DESCRIPTION
**Pull request by mihaelabalutoiu**
This PR adds unit testing for the `events.py` module.

### CODE REVIEW

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolis/tests/test_events.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        self.mock_event_handler.get_progress_update_identifier\
+            .assert_called_once_with(mock.sentinel.progress_update)
+
+        expected_perc_steps = {
+            mock.sentinel.progress_update_id: events._PercStepData(
+                mock.sentinel.progress_update_id, 50, 50, 100)}
+        self.assertEqual(self.event_manager._perc_steps, expected_perc_steps)
+
+    @mock.patch.object(events.LOG, 'warn')
+    def test_add_percentage_step_negative_total_steps(self, mock_log_warn):
```
[the remark this answers is not in the document]
Comment (mihaelabalutoiu) [reply coriolis/tests/test_events.py:57]: > You got some cases to check, why not try using `ddt`? Should make it easier to test all the cases, in one test method.

Good idea, thank you!

**Comment by mihaelabalutoiu** [comment]
@Dany9966 I made the changes, please take a look now. Thank you!

File: coriolis/tests/test_events.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+         'initial_step': 100,
+         'expected_perc': 100,
+         'expected_initial_step': 100,
+         'expected_total_steps': 100},
+        {'message': 'test-message',
+         'total_steps': 100,
+         'initial_step': 50,
+         'expected_perc': 50,
+         'expected_initial_step': 50,
+         'expected_total_steps': 100})
```
Comment (Dany9966) [inline coriolis/tests/test_events.py:70]: Please add a case where `expected_perc` remains 0, i.e. this `if` is skipped:
https://github.com/cloudbase/coriolis/blob/7b1e26f736b8f6a5f7cfb3cf6cfb4dcc728b2657/coriolis/events.py#L55-L57

File: coriolis/tests/test_events.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        self.mock_event_handler.add_progress_update.return_value = (
+            mock.sentinel.progress_update)
+        self.mock_event_handler.get_progress_update_identifier.return_value = (
+            mock.sentinel.progress_update_id)
+
+        self.event_manager.add_percentage_step('test-message', 100, 50)
+        result = self.event_manager.set_percentage_step(
+            self.event_manager._perc_steps[
+                mock.sentinel.progress_update_id], 40)
+        self.assertLogs(events.LOG, logging.WARN)
```
Comment (Dany9966) [inline coriolis/tests/test_events.py:124]: Please either remove this `assertLogs`, or fix its usage

**Review by Dany9966 — CHANGES_REQUESTED** [review]
Mostly LGTM, apart from minor issues

**Pull request merged**